zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -1080,4 +1080,8 @@ private boolean isRunning() {
     private boolean isParticipant(long sid) {
         return self.getVotingView().containsKey(sid);
     }
+
+    public void startSnapshot() {
```

Comment:
I can't find this method in the original patch.
It seems that it is only logging a line.
Is it needed?

Reply:
Actually this method is added to make test can happen.
The issue occurrence must meet several conditions.
One import condition of them is that quorum must commit one proposal during the time window that is between the NEWLEADER package firing and the snapshot firing. To meet it, our test must find a hook to emit one data writing in the time window. 
In 3.5, it invokes getLearnerSnapshotThrottler during the time window. So @lvfangmin original patch is override getLearnerSnapshotThrottler to make a hook to fire the data writing. 
In 3.4, nothing is invoked during the time window. I have not any chance to make a hook. So I add startSnapshot method during the window and use it to make the fire data writing data hook.

 Maybe we can discuss it to make a better one. I have not found it yet.